### PR TITLE
Decode admin proposals in voter dapp

### PR DESCRIPTION
**Pull request by ptare**
Adds a button next to an admin proposal that shows up a dialog explaining the proposal. The explanation text is what the CLI shows: we could try to improve both over time. E.g., interpret the most common admin proposals and show special handling for "setInflationRate".

I tried a few different options, such as a separate table and a tooltip, but this option seemed like the best for now.
Table:
image

Popup:
image

Signed-off-by: Prasad Tare <prasad.tare@gmail.com>

### CODE REVIEW

**Comment by ptare** [comment]
I'm having both prettier and build problems, but some feedback at this stage would be very useful. Mind taking a look? @mrice32 @nicholaspai @chrismaree 

Re the build issue: I can't find a way to suppress the warning in React or in webpack. The only way that works so far is to set `CI=false` even on CI, to make React not error out on warnings.

File: core/scripts/cli/cli_entry.sh [hunk 1]
```diff
@@ -8,7 +8,7 @@ set -e
 # $(npm bin)/uma --network mainnet_ledger
 
 # Platform independent way of evaluating symlinks to find the real location of the script.
-SCRIPT_FILE=$(/usr/bin/env python2 -c "import os; print(os.path.realpath(\"$0\"))")
+SCRIPT_FILE=$(/usr/bin/env python -c "import os; print(os.path.realpath(\"$0\"))")
```
Comment (nicholaspai) [inline core/scripts/cli/cli_entry.sh]: why'd you change this?

Comment (ptare) [reply core/scripts/cli/cli_entry.sh]: It doesn't work on my machine with `python2`.

File: voter-dapp/src/ActiveRequests.js [hunk 2]
```diff
@@ -80,7 +120,9 @@ function ActiveRequests({ votingAccount, votingGateway }) {
     voteStatuses &&
     // Each of the subfetches has to complete. In drizzle, `undefined` means incomplete, while `null` means complete
     // but the fetched value was null, e.g., no `comittedValue` existed.
-    voteStatuses.every(voteStatus => voteStatus.committedValue !== undefined);
+    voteStatuses.every(voteStatus => voteStatus.committedValue !== undefined) &&
+    proposals &&
```
Comment (mrice32) [inline voter-dapp/src/ActiveRequests.js]: Does this mean that we get a brief moment after the user hits "explain" where the page returns to "Looking up requests"?

This isn't a problem or anything, just curious. The only thing better than that might be to pop the dialogue immediately and show loading text there instead. However, I think detecting that exact situation below would be more effort than its worth.

File: common/AdminUtils.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+const adminPrefix = "Admin ";
+
+function isAdminRequest(identifierUtf8) {
+  return identifierUtf8.startsWith(adminPrefix);
+}
+
+// Assumes that `identifierUtf8` is an admin request, i.e., `isAdminRequest()` returns true for it.
+function getAdminRequestId(identifierUtf8) {
+  return parseInt(identifierUtf8.slice(adminPrefix.length), 10);
```
[the remark this answers is not in the document]
Comment (ptare) [reply common/AdminUtils.js]: The radix for parseInt.

File: voter-dapp/src/ActiveRequests.js (hunk 2 above)
Comment (ptare) [reply voter-dapp/src/ActiveRequests.js]: I don't think so actually.

The way I understand it, the page doesn't render until `subsequentFetchComplete` is `true`. The intention is that the the proposals are all fetched from the `Governor` before the page renders at all, but the decoding happens when you press `Explain`.

The hooks stuff is always confusing, so let me know if you think the code doesn't actually behave that like haha.

File: common/AdminUtils.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (mrice32) [reply common/AdminUtils.js]: Oh, I misunderstood -- thought it was an argument to slice. Makes sense.

File: common/AdminUtils.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+  if (!transaction.data || transaction.data.length === 0 || transaction.data === "0x") {
+    // No data -> simple ETH send.
+    returnValue += "\nTransaction is a simple ETH send (no data).";
+  } else {
+    // Txn data isn't empty -- attempt to decode.
+    const decodedTxn = abiDecoder.decodeMethod(transaction.data);
+    if (!decodedTxn) {
+      // Cannot decode txn, just give the user the raw data.
+      returnValue += "\nCannot decode transaction (does not match any UMA Protocol Signature.";
```
Comment (nicholaspai) [inline common/AdminUtils.js]: close these parentheses

File: core/scripts/cli/admin.js [hunk 5]
```diff
@@ -9,33 +10,15 @@ async function decodeGovernorProposal(artifacts, id) {
 
   console.group();
   console.log("Retrieved Admin Proposal for ID:", id);
-  console.log("Proposal has", proposal.transactions.length, " transactions");
+  console.log("Proposal has", proposal.transactions.length, " transaction(s)");
```
Comment (nicholaspai) [inline core/scripts/cli/admin.js]: Totally optional, but could use 

```js
console.log(`Proposal has ${proposal.transactions.length} transaction${proposal.transactions.length === 1 ? '' : 's'}`)
```

File: voter-dapp/src/ActiveRequests.js [hunk 6]
```diff
@@ -63,6 +69,40 @@ function ActiveRequests({ votingAccount, votingGateway }) {
...
+        : null
+    }));
+  });
+  const decodeRequestIndex = index => {
+    if (index === closedDialogIndex) {
+      return "";
+    }
+    const proposal = proposals[index].proposal;
+    let output =
+      hexToUtf8(pendingRequests[index].identifier) + " (" + proposal.transactions.length + " transaction(s))";
```
Comment (nicholaspai) [inline voter-dapp/src/ActiveRequests.js]: did you forget to close these parentheses?

File: common/AdminUtils.js (hunk 4 above)
Comment (ptare) [reply common/AdminUtils.js]: Done

File: core/scripts/cli/admin.js (hunk 5 above)
Comment (ptare) [reply core/scripts/cli/admin.js]: Done

File: voter-dapp/src/ActiveRequests.js (hunk 6 above)
Comment (ptare) [reply voter-dapp/src/ActiveRequests.js]: No, I think this one is correctly closed.

Comment (nicholaspai) [reply voter-dapp/src/ActiveRequests.js]: true

File: voter-dapp/src/ActiveRequests.js (hunk 2 above)
Comment (mrice32) [reply voter-dapp/src/ActiveRequests.js]: Yup, reread, and I was conflating two separate functions in my head. I see what you're saying now. There's one loop that finds all the admin proposals in the list and requests them upfront. And the second function just maps the index to that prefetched data for display synchronously.

Makes sense.

**Comment by ptare** [comment]
Issue #890

**Pull request merged**